src/passwd.c

```diff
@@ -387,10 +384,12 @@ static void check_password (const struct passwd *pw, const struct spwd *sp)
 	 * Passwords may only be changed after sp_min time is up.
 	 */
 	if (sp->sp_lstchg > 0) {
-		time_t ok;
-		ok = (time_t) sp->sp_lstchg * DAY;
-		if (sp->sp_min > 0) {
-			ok += (time_t) sp->sp_min * DAY;
+		long now, ok;
+		now = time (NULL) / DAY;
+		ok = sp->sp_lstchg;
+		if (   (sp->sp_min > 0)
+		    && __builtin_saddl_overflow(ok, sp->sp_min, &ok)) {
+			ok = LONG_MAX;
```

Comment:
I'd prefer the check for overflow in a separate commit.

Reply:
Separated